includes/ucf-email-editor-config.php

```diff
@@ -0,0 +1,253 @@
+				self::$option_prefix . 'header_image',
+				'Header Image',  // formatted field title
+				array( 'UCF_Email_Editor_Config', 'display_settings_field' ), // display callback
+				'ucf_email_editor',  // settings page slug
+				'ucf_email_editor_section_general',  // option section slug
+				array(  // extra arguments to pass to the callback function
+					'label_for'   => self::$option_prefix . 'header_image',
+					'description' => 'URL of the image to be displayed in the header.',
+					'type'        => 'image-url'
+				)
+			);
+
+			add_settings_field(
+				self::$option_prefix . 'president_signature_image',
+				'President Signature Image',  // formatted field title
+				array( 'UCF_Email_Editor_Config', 'display_settings_field' ), // display callback
+				'ucf_email_editor',  // settings page slug
+				'ucf_email_editor_section_general',  // option section slug
+				array(  // extra arguments to pass to the callback function
+					'label_for'   => self::$option_prefix . 'president_signature_image',
+					'description' => 'URL of the image to be displayed in the president\'s email signature.',
+					'type'        => 'image-url'
+				)
+			);
+
+			add_settings_field(
+				self::$option_prefix . 'provost_signature_image',
+				'Provost Signature Image',  // formatted field title
+				array( 'UCF_Email_Editor_Config', 'display_settings_field' ), // display callback
+				'ucf_email_editor',  // settings page slug
+				'ucf_email_editor_section_general',  // option section slug
+				array(  // extra arguments to pass to the callback function
+					'label_for'   => self::$option_prefix . 'provost_signature_image',
+					'description' => 'URL of the image to be displayed in the provost\'s email signature.',
+					'type'        => 'image-url'
+				)
+			);
+		}
+
+
+		/**
+		 * Displays an individual setting's field markup.
+		 **/
+		public static function display_settings_field( $args ) {
+			$option_name   = $args['label_for'];
+			$description   = $args['description'];
+			$field_type    = $args['type'];
+			$hidden        = isset( $args['hidden'] ) ? $args['hidden'] : false;
+			$current_value = self::get_option_or_default( $option_name );
+			$markup        = '';
+
+			switch ( $field_type ) {
+				case 'checkbox':
+					ob_start();
+				?>
+					<input type="checkbox" id="<?php echo $option_name; ?>" name="<?php echo $option_name; ?>" <?php echo ( $current_value == true ) ? 'checked' : ''; ?>>
+					<p class="description">
+						<?php echo $description; ?>
+					</p>
+				<?php
+					$markup = ob_get_clean();
+					break;
+
+				case 'number':
+					ob_start();
+				?>
+					<input type="number" id="<?php echo $option_name; ?>" name="<?php echo $option_name; ?>" value="<?php echo $current_value; ?>">
+					<p class="description">
+						<?php echo $description; ?>
+					</p>
+				<?php
+					$markup = ob_get_clean();
+					break;
+
+				case 'image-url':
+						ob_start();
+					?>
+					<img src="<?php echo $current_value; ?>">
```

Comment:
Could probably wrap this in an if statement so there isn't an image showing if the value is blank or an invalid URL. An if-else might be nice with an error message stating the url is invalid.